### PR TITLE
Bpb 108 cached jobs

### PR DESCRIPTION
**Pull request by DenverM80**
Modify lower Jobs pane in DeepStorageBrowserPresenter:
Instead of a checkbox in the settings, add a toggle button which switches between showing jobs until they are "In Cache" or the Job is "Complete" (persisted to tape)

Still need to address that this behavior can't be changed after the job starts in a separate PR.

### CODE REVIEW

**Comment by DenverM80** [comment]
putjobtocache
putjobtotape

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/DeepStorageBrowserPresenter.java [hunk 1]
```diff
@@ -109,12 +118,12 @@
     private final Ds3Common ds3Common;
     private final JobInterruptionStore jobInterruptionStore;
     private final SettingsStore settingsStore;
+    private ShowCachedJobSettings showCachedJobSettings;
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/DeepStorageBrowserPresenter.java]: Is this reassigned?  If not it should be made final

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/DeepStorageBrowserPresenter.java [hunk 2]
```diff
@@ -335,7 +388,19 @@ private void logsORJobsMenuItemAction(final CheckMenuItem menuItem, final VBox v
...
+    private void registerLoggingServiceListener() {
+        LOG.info("Registering loggingService observable");
+        final Observable<LoggingService.LogEvent> observable = loggingService.getLoggerObservable();
+        observable.doOnNext(logEvent -> {
+            if (Platform.isFxApplicationThread()) {
+                this.logTextForParagraph(logEvent.getMessage(), logEvent.getLogType());
+            } else {
+                Platform.runLater(() -> this.logTextForParagraph(logEvent.getMessage(), logEvent.getLogType()));
+            }
+        }).subscribe();
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/DeepStorageBrowserPresenter.java]: Was this code block always like this?  Even if it was, we should change this.

JavaRX allows you to tell it which thread you want the handlers to run in when fired.  We should update this to so that it tells it to run in the fxapplicationthread, then we don't need to do this application thread check.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3PutJob.java [hunk 3]
```diff
@@ -299,23 +297,27 @@ public String getDs3ObjectPath(final Path key, final Path value, final boolean c
...
+
         if (isCacheJobEnable) {
-            updateProgress(totalJobSize, totalJobSize);
-            updateMessage(StringBuilderUtil.jobSuccessfullyTransferredString(JobRequestType.PUT.toString(), FileSizeFormat.getFileSizeType(totalJobSize), bucket + "\\" + targetDir, dateOfTransfer, resourceBundle.getString("blackPearlCache"), isCacheJobEnable).toString());
-            updateProgress(totalJobSize, totalJobSize);
-            loggingService.logMessage(StringBuilderUtil.jobSuccessfullyTransferredString(JobRequestType.PUT.toString(), FileSizeFormat.getFileSizeType(totalJobSize), bucket + "\\" + targetDir, dateOfTransfer, resourceBundle.getString("blackPearlCache"), isCacheJobEnable).toString(), LogType.SUCCESS);
-            GetJobSpectraS3Response response = ds3Client.getJobSpectraS3(new GetJobSpectraS3Request(jobId));
-            while (!response.getMasterObjectListResult().getStatus().toString().equals(StringConstants.JOB_COMPLETED)) {
+            while (!ds3Client.getJobSpectraS3(new GetJobSpectraS3Request(jobId)).getMasterObjectListResult().getStatus().toString().equals(StringConstants.JOB_COMPLETED)) {
                 Thread.sleep(60000);
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3PutJob.java]: What's the point of this sleep?

Comment (DenverM80) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3PutJob.java]: it looks like a 5 minute polling cycle waiting for status COMPLETE.  My only change here was to pull out duplicated log / status messages

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/DeepStorageBrowserPresenter.java (hunk 2 above)
Comment (DenverM80) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/DeepStorageBrowserPresenter.java]: will fix.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/DeepStorageBrowserPresenter.java (hunk 1 above)
Comment (DenverM80) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/DeepStorageBrowserPresenter.java]: will fix

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3PutJob.java (hunk 3 above)
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3PutJob.java]: We should look at seeing if we can find a way to refactor this to remove the sleep.  In the sea vault code we used a recurring task to do something like this.  The main reason I'm not a fan of this kind of sleep is that we are consuming a whole thread while this is happening, rather than freeing the thread to do something else.

Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3PutJob.java]: We can do this in a different pull request though

Comment (DenverM80) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3PutJob.java]: I agree there's some "code smell" in this polling cycle for the reasons you mentioned.  I'll create a separate task to track this.

**Comment by DenverM80** [comment]
updated
